torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -727,32 +730,60 @@ def _shard_parameters(self) -> None:
             self.params
         ), "numel_padded_per_param is not populated correctly."
 
-    def _get_shard(self, tensor: torch.Tensor) -> Tuple[torch.Tensor, int]:
-        """Return the local shard of a full tensor."""
-        # Shard using torch.chunk to match all-gather/reduce-scatter.
-        chunks = torch.flatten(tensor).chunk(self.world_size)
-        if len(chunks) < (self.rank + 1):
+    @staticmethod
```

Comment:
`_get_chunk()` is separated since it includes all the logic up to but not including the point where new tensor storage may be allocated. This is needed to compute the sharded tensor shape without allocating new storage.